### PR TITLE
Update incorrect comments and add new comments about record IDs

### PR DESCRIPTION
**Pull request by mhammond**
Sammy and I chatted a little about this in the context of #5277. This will end up causing a (hopefully trivial) conflict in that PR, but  felt it was easier than trying to ask for the comments to be added in that PR.

### CODE REVIEW

**Comment by mhammond** [comment]
(please merge this once approved, then rebase your PR on top of it keeping these new comments)

File: components/tabs/src/sync/engine.rs
Comment (skhamis) [inline components/tabs/src/sync/engine.rs:60]: nit: `which it must be`

File: components/tabs/src/storage.rs
Comment (skhamis) [inline components/tabs/src/storage.rs:35]: I was half tempted to just rename this to prevent confusion, but it's used quite a bit in the clients -- so the comment will be perfect for now

**Review by skhamis — APPROVED** [review]
Small nit but I'll just fix it in my PR -- thanks for adding this! Definitely clears up some details about how the client_id is actually being used.

**Pull request merged**
